### ISSUE TITLE
[DataGrid] Improve toolbar documentation

### ISSUE THREAD

**Issue by 19007361** [mui-org/material-ui-x#1463]
I cannot seem to find any of the options that GridToolbar can take other than looking directly in the source code and reverse engineering what they're supposed to do, e.g. "disableDensitySelector" which is nowhere in the documentation.
I am not specifically referring to just this, there are many other components (especially in Data Grid) where the API simply links to <DataGrid /> API and doesn't include the other components mentioned.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-x/issues) of this repository and believe that this is not a duplicate.

## Steps to Reproduce 🕹

Steps:

1.
2.
3.
4.

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
  Don't forget to mention which browser you used.
  Output from `npx @material-ui/envinfo` goes here.
```
</details>

**Comment by oliviertassinari** [comment]
@19007361 Thanks for the feedback. We have two issues related to the generation of the API, which is meant to be comprehensive: #1213, #1211. So in its current, we can't leverage your feedback, as we are already aware of the issue, as well as how important it's (it's one of the most important problems we could currently work on). 

I think that we could continue the discussion on the existing content. Would it help if we list all the toolbar components available in https://material-ui.com/components/data-grid/components/#toolbar?

**Comment by 19007361** [comment]
Your response has been very helpful, thank you.

Yes, that would really help, as I'm assuming through these issues you linked that the props of each toolbar changes regularly and therefore you cannot reliably describe each toolbar element's props in the API.

**Comment by dtassone** [comment]
The components documentation will be improved as part of #1382

**Comment by oliviertassinari** [comment]
@dtassone Could we break down #1382? It's currently mostly empty. We could focus on 1. toolbar, then 2. column menu, 3. then custom icon, etc.

**Comment by dtassone** [comment]
I have now fixed #1382, and we also move the density doc in the accessibility page 
There is `disableDensitySelector ` mentioned as well so closing this issue for now.
Please open a new issue, if a specific component or anything is missing.
